src/Orleans/Core/GrainFactory.cs

```diff
@@ -56,14 +62,13 @@ internal GrainFactory()
         /// <typeparam name="TGrainInterface">The interface to get.</typeparam>
         /// <param name="primaryKey">The primary key of the grain.</param>
         /// <param name="grainClassNamePrefix">An optional class name prefix used to find the runtime type of the grain.</param>
-        /// <returns></returns>
+        /// <returns>A reference to the specified grain.</returns>
         public TGrainInterface GetGrain<TGrainInterface>(Guid primaryKey, string grainClassNamePrefix = null) where TGrainInterface : IGrainWithGuidKey
         {
-            return Cast<TGrainInterface>(
-                GrainFactoryBase.MakeGrainReference_FromType(
-                    baseTypeCode => TypeCodeMapper.ComposeGrainId(baseTypeCode, primaryKey, typeof(TGrainInterface)),
-                    typeof(TGrainInterface),
-                    grainClassNamePrefix));
+            Type interfaceType = typeof(TGrainInterface);
+            var implementation = this.GetGrainClassData(interfaceType, grainClassNamePrefix);
```

Comment:
As you stated, this makes PR's harder to read because we don't have the 'tooling' in the review process.  Having said that, this is just a readability issues, so I'll not push the issue.  Change if you feel inclined, or leave as is..  Main goal for me was to communicate the (unofficial) standard so you were aware of it in future changes.